// @ionic-native/date-picker/__ivy_ngcc__/ngx/index.js
import { __extends } from "tslib";
import { Injectable } from '@angular/core';
import { IonicNativePlugin, cordova } from '@ionic-native/core';
import * as ɵngcc0 from '@angular/core';
var DatePicker = /** @class */ (function (_super) {
    __extends(DatePicker, _super);
    function DatePicker() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        /**
         * @hidden
         */
        _this.ANDROID_THEMES = {
            THEME_TRADITIONAL: 1,
            THEME_HOLO_DARK: 2,
            THEME_HOLO_LIGHT: 3,
            THEME_DEVICE_DEFAULT_DARK: 4,
            THEME_DEVICE_DEFAULT_LIGHT: 5,
        };
        return _this;
    }
    DatePicker.prototype.show = function (options) { return cordova(this, "show", {}, arguments); };
    DatePicker.pluginName = "DatePicker";
    DatePicker.plugin = "cordova-plugin-datepicker";
    DatePicker.pluginRef = "datePicker";
    DatePicker.repo = "https://github.com/VitaliiBlagodir/cordova-plugin-datepicker";
    DatePicker.platforms = ["Android", "iOS", "Windows"];
DatePicker.ɵfac = /*@__PURE__*/ function () { var ɵDatePicker_BaseFactory; return function DatePicker_Factory(t) { return (ɵDatePicker_BaseFactory || (ɵDatePicker_BaseFactory = ɵngcc0.ɵɵgetInheritedFactory(DatePicker)))(t || DatePicker); }; }();
DatePicker.ɵprov = /*@__PURE__*/ ɵngcc0.ɵɵdefineInjectable({ token: DatePicker, factory: function (t) { return DatePicker.ɵfac(t); } });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(DatePicker, [{
        type: Injectable
    }], null, null); })();
    return DatePicker;
}(IonicNativePlugin));
export { DatePicker };
